src/utils/encoreCatalogLogOutTimer.js

```diff
@@ -25,37 +32,38 @@ function EncoreLogOutTimer() {
         utils.deleteCookie('nyplIdentityPatron');
       }
 
-      // Delete cookie "ENCORE_LAST_VISITED" which holds the last time the user visited Encore
-      // if the cookie "PAT_LOGGED_IN" does not exist
-      if (utils.hasCookie('ENCORE_LAST_VISITED')) {
-        utils.deleteCookie('ENCORE_LAST_VISITED');
+      // Delete cookie "VALID_DOMAIN_LAST_VISITED" which holds the last time the user visited
+      // the valid domain if the cookie "PAT_LOGGED_IN" does not exist
+      if (utils.hasCookie('VALID_DOMAIN_LAST_VISITED')) {
+        utils.deleteCookie('VALID_DOMAIN_LAST_VISITED');
       }
 
       // Completely log out the user
       this.loadLogoutIframe(isTest);
     } else {
-      if (isOnEncore) {
-        // Set the cookie "ENCORE_LAST_VISITED" once the user visited Encore
-        utils.setCookie('ENCORE_LAST_VISITED', currentTime);
-        this.logOutFromEncoreIn(encoreLogInExpireDuration, isTest);
+      if (isOnValidDomain) {
+        // Set the cookie "VALID_DOMAIN_LAST_VISITED" once the user visited Encore or Catalog
+        utils.setCookie('VALID_DOMAIN_LAST_VISITED', currentTime);
+        this.logOutFromEncoreAndCatalogIn(encoreLogInExpireDuration, isTest);
       } else {
-        const lastVisitedEncoreTime = utils.getCookie('ENCORE_LAST_VISITED');
-        const timeTillLogOut = lastVisitedEncoreTime
-          ? encoreLogInExpireDuration - (currentTime - lastVisitedEncoreTime)
+        const lastVisitedValidDomainTime = utils.getCookie('VALID_DOMAIN_LAST_VISITED');
+        const timeTillLogOut = lastVisitedValidDomainTime
+          ? encoreLogInExpireDuration - (currentTime - lastVisitedValidDomainTime)
           : undefined;
```

Comment:
Why undefined instead of 0?

Reply:
I did this because it is not really 0 seconds, which comes from count down. The final logout timing will be the same based on line 67, but it will be more useful for a developer to log or debug.

For example, if we get an `undefined` then we know that there is no valid `lastVisitedValidDomainTime`. While if we always got 0 as a fallback, then we have to figure out if it is from no valid `lastVisitedValidDomainTime` or because the count down is just finished.